lib/cartopy/mpl/geoaxes.py

```diff
@@ -1276,14 +1276,29 @@ def imshow(self, img, *args, **kwargs):
             regrid_shape = self._regrid_shape_aspect(regrid_shape,
                                                      target_extent)
             warp_array = cartopy.img_transform.warp_array
+            unmodified_extent = extent
             img, extent = warp_array(img,
                                      source_proj=transform,
-                                     source_extent=extent,
+                                     source_extent=unmodified_extent,
                                      target_proj=self.projection,
                                      target_res=regrid_shape,
                                      target_extent=target_extent,
                                      mask_extrapolated=True,
                                      )
+            alpha = kwargs.pop('alpha', None)
+            alpha_arr = np.array(alpha)
+            if alpha_arr.ndim == 2:
```

Comment:
Could you put `if np.array(alpha).ndim == 2:` here and then just keep using alpha below. Then you wouldn't need the `else` statement at all and could just set `kwargs['alpha'] = alpha` after the if statement.

Reply:
OK---modified in latest commit.